src/client/driver/command-executors/execute-selector.js

```diff
@@ -0,0 +1,31 @@
+import SelectorExecutor from './client-functions/selector-executor';
+
+export function getResult (command, globalTimeout, startTime, createNotFoundError, createIsInvisibleError, statusBar) {
+    var selectorExecutor = new SelectorExecutor(command, globalTimeout, startTime, createNotFoundError, createIsInvisibleError);
+
+    statusBar.showWaitingElementStatus(selectorExecutor.timeout);
+
+    return selectorExecutor.getResult()
+        .then(el => {
+            return statusBar.hideWaitingElementStatus(!!el)
+                .then(() => el);
+        })
+        .catch(err => {
+            return statusBar.hideWaitingElementStatus(false)
+                .then(() => {
+                    throw err;
+                });
+        });
+}
+
+export function getResultDriverStatus (command, globalTimeout, startTime, createNotFoundError, createIsInvisibleError, statusBar) {
+    var selectorExecutor = new SelectorExecutor(command, globalTimeout, startTime, createNotFoundError, createIsInvisibleError);
+
+    statusBar.showWaitingElementStatus(selectorExecutor.timeout);
+
+    return selectorExecutor.getResultDriverStatus()
+        .then(status => {
+            return statusBar.hideWaitingElementStatus(!!status.result && !status.executionError)
```

Comment:
Why we should not hide statusBar in case of `status.executionError`? Also if we have `status.executionError` we always don't have `status.result`.